cmd/ib-sriov-cni/main.go

```diff
@@ -320,10 +360,7 @@ func cmdDel(args *skel.CmdArgs) (retErr error) {
 	sm := sriov.NewSriovManager()
 
 	if netConf.IPAM.Type != "" {
-		if netConf.IPAM.Type == ipamDHCP {
-			return fmt.Errorf("ipam type dhcp is not supported")
-		}
-		err = ipam.ExecDel(netConf.IPAM.Type, args.StdinData)
+		err = handleIPAMCleanup(netConf, args.StdinData)
```

Comment:
do we really need this refactor now ? i see this is only used here.

fine to keep as is if u prefer though.

one note, as @SchSeba  added above, lets move the check for vfio pci mode here and invoke cleanup func conditionally